src/shogun/labels/MulticlassLabels.cpp

```diff
@@ -219,6 +215,8 @@ namespace shogun
 				    orig->as<CMulticlassLabels>());
 			case LT_DENSE_GENERIC:
 				return to_multiclass(orig->as<CDenseLabels>());
+			case LT_BINARY:
+				return some<CMulticlassLabels>(orig->as<CBinaryLabels>());
```

Comment:
This constructor is duplicate code with the thing you wrote above, could you unify that?